pkg/providers/awskms/awskms.go

```diff
@@ -0,0 +1,61 @@
+package awskms
+
+import (
+	"encoding/base64"
+
+	"github.com/variantdev/vals/pkg/api"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/variantdev/vals/pkg/awsclicompat"
+)
+
+type provider struct {
+	// Keeping track of KMS services since we need a service per region
+	client *kms.KMS
+
+	// AWS KMS global configuration
+	Region, Profile string
+}
+
+func New(cfg api.StaticConfig) *provider {
+	p := &provider{}
+	p.Region = cfg.String("region")
+	p.Profile = cfg.String("profile")
+	return p
+}
+
+func (p *provider) GetString(key string) (string, error) {
+	cli := p.getClient()
+
+	blob, err := base64.StdEncoding.DecodeString(key)
+	if err != nil {
+		return "", err
+	}
+
+	in := &kms.DecryptInput{
+		CiphertextBlob: blob,
+	}
+
+	result, err := cli.Decrypt(in)
+	if err != nil {
+		return "", err
+	}
+
+	return string(result.Plaintext), nil
+}
+
+func (p *provider) GetStringMap(key string) (map[string]interface{}, error) {
```

Comment:
It would be great if you could implement this too!
FYI, `GetStringMap` can be implemented on top of `GetString` like seen in https://github.com/variantdev/vals/blob/e059ff4fa22925a54038c69a62c51ee500d1c4d2/pkg/providers/s3/s3.go#L71-L82.
It's used when you instructed `vals` to parse the decrypted/fetched value as a YAML or a JSON object and retrieve only one value within the object (e.g. when you suffixed the URL with `#/yaml_or_json_key/in/secret`

Reply:
I do think it's fairly unlikely that anyone would use the KMS provider in this manner; since the ciphertext needs to be embedded directly into the document (and embedded repeatedly if you need to reference it multiple times), you'd usually just encode the value of the specific value needed at each location rather than encoding a much larger document (which means a correspondingly larger cipher text), and repeating that ciphertext in multiple places with different keys.

Nevertheless, I did go ahead and implement it, so if someone _does_ want to use it this way, they can.

Reply:
That's definitely a valid point! Thanks for your insight. I'm still glad you made it feature-parity with other providers so that it is at least consistent across the providers and users might potentially find a way to utilize it in practice.